File: scripts/parity/scenarios/py/viz_basic_line.py
import fullcontrol as fc
import json

# Simple line visualization test - matches basic_line but with plot output
printer = fc.Printer(print_speed=1800, travel_speed=6000)
extruder = fc.Extruder(units='mm', dia_feed=1.75, relative_gcode=False, travel_format='G1_E0')
geom = fc.ExtrusionGeometry(area_model='rectangle', width=0.45, height=0.2)

seq = [
    printer,
    extruder,
    geom,
    fc.Point(x=0, y=0, z=0.2),
    fc.Extruder(on=True),
    fc.Point(x=10, y=0, z=0.2),
    fc.Extruder(on=False)
]

# Get raw plot data
plot_data = fc.transform(seq, result_type='plot', controls=fc.PlotControls(raw_data=True))

# Convert to JSON-serializable format
output = {
    'paths': [],
    'boundingBox': {
        'minx': plot_data.bounding_box.minx,
        'maxx': plot_data.bounding_box.maxx,
        'midx': plot_data.bounding_box.midx,
        'rangex': plot_data.bounding_box.rangex,
        'miny': plot_data.bounding_box.miny,
        'maxy': plot_data.bounding_box.maxy,
        'midy': plot_data.bounding_box.midy,
        'rangey': plot_data.bounding_box.rangey,
        'minz': plot_data.bounding_box.minz,
        'maxz': plot_data.bounding_box.maxz,
        'midz': plot_data.bounding_box.midz,
        'rangez': plot_data.bounding_box.rangez
    },
    'annotations': plot_data.annotations
}

for path in plot_data.paths:
    output['paths'].append({
        'xvals': path.xvals,
        'yvals': path.yvals,
        'zvals': path.zvals,
        'colors': path.colors,
        'widths': path.widths,
        'heights': path.heights,
        'extruder': {'on': path.extruder.on}
    })

print(json.dumps(output, indent=2))

File: scripts/parity/scenarios/js/viz_basic_line.mjs
// Visualization test: simple line matching Python version
import { Point, Printer, Extruder, ExtrusionGeometry, transform } from '../../../../dist/index.js'

const printer = new Printer({ print_speed: 1800, travel_speed: 6000 })
const extruder = new Extruder({ units: 'mm', dia_feed: 1.75, relative_gcode: false, travel_format: 'G1_E0' })
const geom = new ExtrusionGeometry({ area_model: 'rectangle', width: 0.45, height: 0.2 })

const seq = [
  printer,
  extruder,
  geom,
  new Point({ x:0, y:0, z:0.2 }),
  new Extruder({ on:true }),
  new Point({ x:10, y:0, z:0.2 }),
  new Extruder({ on:false })
]

// Get raw plot data
const result = transform(seq, 'plot', { raw_data: true, show_tips: false, show_banner: false })
const plotData = result.plot

// Convert to JSON output
const output = plotData.toJSON()
process.stdout.write(JSON.stringify(output, null, 2) + '\n')
